### PR TITLE
inclusion of all currently implemented layers in layer.hpp

### PR DESCRIPTION
**Pull request by mrityunjay-tripathi**

### CODE REVIEW

File: src/mlpack/methods/ann/layer/layer.hpp [hunk 1]
```diff
@@ -12,32 +12,57 @@
...
+#include "concatenate.hpp"
+#include "constant.hpp"
 #include "convolution.hpp"
 #include "dropconnect.hpp"
+#include "dropout.hpp"
+#include "elu.hpp"
+#include "fast_lstm.hpp"
+#include "flexible_relu.hpp"
 #include "glimpse.hpp"
+#include "gru.hpp"
```
Comment (birm) [inline src/mlpack/methods/ann/layer/layer.hpp:35]:
```suggestion
```

File: src/mlpack/methods/ann/layer/layer.hpp [hunk 2]
```diff
@@ -12,32 +12,57 @@
...
+#include "select.hpp"
 #include "sequential.hpp"
+#include "softshrink.hpp"
 #include "subview.hpp"
-#include "concat.hpp"
-#include "vr_class_reward.hpp"
 #include "transposed_convolution.hpp"
+#include "virtual_batch_norm.hpp"
+#include "vr_class_reward.hpp"
 #include "weight_norm.hpp"
```
Comment (birm) [inline src/mlpack/methods/ann/layer/layer.hpp:67]:
```suggestion

#include "gru.hpp"
```

**Comment by mrityunjay-tripathi** [comment]
>I'm not sure why this is so yet, but moving gru to the bottom fixed my build locally.

@birm: Dependencies?

**Comment by mrityunjay-tripathi** [comment]
yes, ```gru.hpp``` has to be after ```linear_no_bias.hpp``` 👍

**Comment by rcurtin** [comment]
> yes, `gru.hpp` has to be after `linear_no_bias.hpp` +1

Maybe we should include `linear_no_bias.hpp` in `gru.hpp`?

**Comment by mrityunjay-tripathi** [comment]
@rcurtin: Oo, my head is banging in this nexus. Now what I see is that ```gru.hpp``` uses other layers as well but doesn't explicitly include them, because inclusion of ```layer_types.hpp``` does that. I tested by including ```linear_no_bias.hpp``` in ```layer_types.hpp``` (since it is not already included in the current implementation) and placed ```gru.hpp``` alphabetically in ```layer.hpp``` and it builds successfully on Linux. What do you say?

**Comment by rcurtin** [comment]
@mrityunjay-tripathi that looks like a good solution to me. :+1:

**Pull request merged**